sgt-back-end/index.js

```diff
@@ -0,0 +1,135 @@
+const pg = require('pg');
+const db = new pg.Pool({
+  connectionString: 'postgres://dev:dev@localhost/studentGradeTable',
+  ssl: {
+    rejectUnauthorized: false
+  }
+});
+const express = require('express');
+const app = express();
+
+app.use(express.json());
+
+app.get('/api/grades', (req, res) => {
+  const sql = `
+    SELECT *
+    FROM "grades"
+  `;
+  db.query(sql)
+    .then(result => {
+      const grade = result.rows;
+      res.status(200).json(grade);
+    })
+    .catch(err => {
+      console.error(err);
+      res.status(500).json({
+        error: 'An unexpected error occured.'
+      });
+    });
+});
+
+app.post('/api/grades', (req, res) => {
+  const name = req.body.name;
+  const course = req.body.course;
+  const score = Number(req.body.score);
+  if (!name || !course || !Number.isInteger(score) || score < 0 || score > 100) {
+    res.status(400).json({
+      error: 'Name, course, and score are required fields. Score must be an integer between 0 and 100'
+    });
+  } else {
+    const sql = `
+      INSERT INTO "grades" ("name", "course", "score")
+      VALUES ($1, $2, $3)
+      RETURNING *;
+    `;
+    const params = [name, course, score];
+    db.query(sql, params)
+      .then(result => {
+        const grade = result.rows[0];
+        res.status(201).json(grade);
+      })
+      .catch(err => {
+        console.error(err);
+        res.status(500).json({
+          error: 'An unexpected error occured.'
+        });
+      });
+  }
+});
+
+app.put('/api/grades/:gradeId', (req, res) => {
+  const name = req.body.name;
+  const course = req.body.course;
+  const score = Number(req.body.score);
+  const gradeId = Number(req.params.gradeId);
+  if (!Number.isInteger(gradeId) || gradeId < 0 || !name || !course || !Number.isInteger(score) || score < 0 || score > 100) {
```

Comment:
`0` should not be a counted as a valid `gradeId` (database `serial` fields start at `1`).